### ISSUE TITLE
Define a context file

### ISSUE THREAD

**Issue by cboettig** [cboettig/nexld#1]
Need a context file that defines types appropriately, particularly so that any references are typed as `@id` rather than the default `@value`.  This way, JSON-LD knows when it sees things like `"otus": "tax1"` that `tax1` is an `id`.  (So that users don't have to write out: `"otus": { "@id": "tax1"}`, which gets verbose and ugly quickly.

**Comment by cboettig** [comment]
We have a context file now: [nexml.json](https://github.com/cboettig/nexld/blob/master/inst/context/nexml.json)

Also have a souped-up context file that adds a bunch of common namespaces, including those on treebase, called [treebase.json](https://github.com/cboettig/nexld/blob/master/inst/context/treebase.json).  This means that most nexld documents can simply use this URL as the context rather than add all namespaces used in various meta elements.  

Looks pretty simple, hopefully I didn't miss any attributes that are ids.  However, the impact of this is not simple: once JSON-LD knows all these things are references, the document is a lot more liquid and we'll need a frame to make sure we can dictate when a property is just an `id` and when it should embed.

Meanwhile, leaving this open at least until we establish a more standardized URL for the the context file. 

My plan is to tag the current git release, and then create a PURL for the context file along the lines of `http://purl.org/nexml/1.0`?  Maybe `http://purl.org/nexml/context` ? Thoughts / preferences @hlapp @rvosa ?

**Comment by rvosa** [comment]
I think it should have some sort of version in it (so, rather
http://purl.org/nexml/1.0 than http://purl.org/nexml/context), though I
would not be so bold to jump for 1.0 right away. The NeXML schema is still
at 0.9 :-)

**Comment by cboettig** [comment]
Sounds good.  I'll use a version-based URL, starting at `0.1`.  (I thought about tying it to the NeXML version string, but then at least at the start our context might be more volatile).  

It's nice to have a concise URL, but I think we should indicate this a `context` file as opposed to something else, so maybe:

- http://purl.org/nexml/0.1/context

This way we can distinguish it from other things we might also want to provide, e.g. a default json-ld frame as <http://purl.org/nexml/0.1/frame>, which is needed to cast arbitrary nexld into the structure required by NeXML. (Unfortunately a frame is agnostic about node ordering, so that would have to be fixed in some other way.)

**Comment by rvosa** [comment]
Yeah, it makes sense to specify it's a context. Is the order right or
should it be //context/0.1?

(Sorry if I seem to second guess all of this - I'm just thinking out loud
so we don't settle on something we'll regret down the line when we can't
change it anymore...)

**Comment by cboettig** [comment]
right, makes more sense to put the version on the end, thanks.

**Comment by cboettig** [comment]
it's alive! 👽 

http://purl.org/nexml/context/0.1
